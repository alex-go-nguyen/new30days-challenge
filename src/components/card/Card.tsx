import React, { PropsWithChildren } from 'react';
import classNames from 'classnames/bind';
import styles from './Card.module.scss';
const cx = classNames.bind(styles);

export interface CardProps extends PropsWithChildren {
    title: string;
}

export default function Card({ title, children }: CardProps) {
    return (
        <div className={cx('card')}>
            <p className={cx('title')}>{title}</p>
            <p className={cx('content')}>{children}</p>
        </div>
    );
}
