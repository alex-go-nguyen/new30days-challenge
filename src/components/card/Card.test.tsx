import React from 'react';
import renderer, { ReactTestRendererJSON } from 'react-test-renderer';
import Card from './Card';

describe('Card component', () => {
    test('should render correctly', () => {
        const component = renderer.create(<Card title="Notification">You are the best!</Card>);
        let tree = component.toJSON() as ReactTestRendererJSON;
        expect(tree).toMatchSnapshot();
    });
});
