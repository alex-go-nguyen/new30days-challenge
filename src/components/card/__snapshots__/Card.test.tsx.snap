// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component should render correctly 1`] = `
<div
  className="card"
>
  <p
    className="title"
  >
    Notification
  </p>
  <p
    className="content"
  >
    You are the best!
  </p>
</div>
`;
